### PR TITLE
Improve our timestamp handling for our sessions

### PR DESCRIPTION
**Pull request by poljar**
This PR mainly switches from an `Instant` to the Ruma type `SecondsSinceUnixEpoch`, the type, while not being fully WASM compatible, has the advantage that it can be easily serialized and deserialized.

Other E2EE implementations, i.e. the EA one, use Unix timestamps as well, so this eases up the migration as well.

### CODE REVIEW

**Comment by jplatte** [comment]
I think it makes sense to have a `"js"` feature in Ruma by the way, that does the same as the wasm support in the `instant` crate. It has to be a separate feature instead of just target-specific behavior though, as compiling to wasm does not necessarily mean that you're targeting a browser environment (with the current 'unknown' target triple one has to use).

File: crates/matrix-sdk-crypto/src/backups/keys/recovery.rs [hunk 1]
```diff
@@ -122,7 +122,8 @@ impl RecoveryKey {
         let decoded = Zeroizing::new(crate::utilities::decode(key)?);
 
         if decoded.len() != Self::KEY_SIZE {
-            Err(DecodeError::Length(decoded.len(), Self::KEY_SIZE))
+            println!("{:?}", decoded);
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/backups/keys/recovery.rs:125]: debug println?

Comment (poljar) [reply crates/matrix-sdk-crypto/src/backups/keys/recovery.rs:125]: Heh, indeed.

File: crates/matrix-sdk-crypto/src/olm/group_sessions/outbound.rs [hunk 2]
```diff
@@ -118,7 +117,7 @@ pub struct OutboundGroupSession {
     account_identity_keys: Arc<IdentityKeys>,
     session_id: Arc<str>,
     room_id: Arc<RoomId>,
-    pub(crate) creation_time: Arc<Instant>,
+    pub(crate) creation_time: Arc<SecondsSinceUnixEpoch>,
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/olm/group_sessions/outbound.rs:120]: Why is this in an `Arc`? It's cheaper to copy the `SecondsSinceUnixEpoch` (and I'm sure the same was true of `Instant`).

**Review by jplatte — APPROVED** [review]
One more nitpick, otherwise LGTM.

**Comment by poljar** [comment]
> I think it makes sense to have a `"js"` feature in Ruma by the way, that does the same as the wasm support in the `instant` crate. It has to be a separate feature instead of just target-specific behavior though, as compiling to wasm does not necessarily mean that you're targeting a browser environment (with the current 'unknown' target triple one has to use).

That would indeed be nice, if for nothing else, to move the special constructors we now have in the rust-sdk.

**Pull request merged**